docs/DeveloperGuide.md

```diff
@@ -1,38 +1,516 @@
+- `Command`: Executes the command requested by the user.
+- `Data`: Handles the data created by the user.
+
+The `Logger` component handles the logging of events, and writes logs to a `logs.txt` log file.
+
+The user primarily interacts with the `Ui` component.
+In turn, the `Parser` interprets the user's input and creates the appropriate `Command`.
+The `Data` component then manages all data created by the user upon `Command` execution.
+The `Storage` component handles the storage and loading of saved user data.
+
+#### How the architecture components interact with each other
+
+The **Sequence Diagram** below shows how the components interact with each other for the scenario where the user issues the command `list`.
+
+![Sequence Diagram](images/OverallSequenceDiagram.png)
+
+1. User enters the command `list` to the `Ui`.
+2. `Ui` passes the command as a string through the method `readUserInput('list')`, called via `BinBash`.
+3. `BinBash` passes the string to `Parser` through the method `parseCommand('list')`.
+4. `Parser` returns a new `Command` object. (In this specific case, it would be a ListCommand object)
+5. `BinBash` calls the `execute()` method of `Command`.
+6. `Command` then interacts with `ItemList`, and calls the relevant method.
+7. `ItemList` returns the executionUiOutput, in the form of a String object. 
+8. `BinBash` calls the `getExecutionUiOutput` command in `Command`
+9. `Command` returns the outputString, in the form of a String object.
+10. `BinBash` calls the `talk()` method in `Ui`, and passes the outputString.
+11. `Ui` prints this outputString to the user.
+12. If the `Command` executed modifies the database, `BinBash` will call the `saveToStorage()` method of `Storage`
+
+### Ui Component
+
+![Ui class diagram](images/UiClassDiagram.png)
+
+API: [`Ui.java`](https://github.com/AY2324S2-CS2113T-T09-2/tp/blob/master/src/main/java/seedu/binbash/ui/Ui.java)
+
+The above class diagram shows the components delegating separate functionalities of the Ui.
+
+The `TextIn` class is responsible for reading user input and returning it to `Ui`, upon which it is passed to the *main()* program.
+
+The `PrintStream` class writes text at the behest of `Ui` to standard output, upon which it is received by the user.
+
+Note the use of an externally provided `LineReader` object in the `TextIn` class that handles input. This allows us to greatly extend our text-based user interface with features such as:
+
+1. Command completion on tab
+
+2. Displaying option descriptions on hover
+
+3. Contextual help menus
+
+![linereader](images/ui-linereader-enhancement.png)
+
+This allows us to overload options on a small number of commands to provide full functionality of the application. Developers can then extend its features without also the worry of finding a way for users to access those features easily.
+
+### Storage Component
+
+![Storage class diagram](images/StorageClassDiagram.png)
```

Comment:
Consider using characters instead of icons for the access modifiers.
![Screenshot 2024-04-04 152440](https://github.com/nus-cs2113-AY2324S2/tp/assets/110762696/17396bf5-2c17-42a9-aace-35b5f04b5145)